packages/backend/src/nest/registration/registration.service.ts

```diff
@@ -1,178 +1,68 @@
-import { Inject, Injectable, OnModuleInit } from '@nestjs/common'
-import express from 'express'
-import getPort from 'get-port'
-import { Agent, Server } from 'http'
+import { Injectable, OnModuleInit } from '@nestjs/common'
 import { EventEmitter } from 'events'
-import {
-  registerOwner,
-  registerUser,
-  RegistrarResponse,
-  RegistrationResponse,
-  sendCertificateRegistrationRequest,
-} from './registration.functions'
-import {
-  ConnectionProcessInfo,
-  ErrorCodes,
-  ErrorMessages,
-  LaunchRegistrarPayload,
-  PermsData,
-  RegisterOwnerCertificatePayload,
-  SocketActionTypes,
-} from '@quiet/types'
-import { EXPRESS_PROVIDER } from '../const'
+import { extractPendingCsrs, issueCertificate } from './registration.functions'
+import { ErrorCodes, ErrorMessages, PermsData, RegisterOwnerCertificatePayload, SocketActionTypes } from '@quiet/types'
 import { RegistrationEvents } from './registration.types'
-import { ServiceState } from '../connections-manager/connections-manager.types'
 import Logger from '../common/logger'
 
 @Injectable()
 export class RegistrationService extends EventEmitter implements OnModuleInit {
   private readonly logger = Logger(RegistrationService.name)
-  private _server: Server
-  private _port: number
-  public registrationService: any
   public certificates: string[] = []
   private _permsData: PermsData
-  private _ownerCertificate: string
 
-  constructor(@Inject(EXPRESS_PROVIDER) public readonly _app: express.Application) {
+  constructor() {
     super()
   }
 
   onModuleInit() {
-    this.on(RegistrationEvents.SET_CERTIFICATES, certs => {
-      this.setCertificates(certs)
-    })
-    this.on(RegistrationEvents.REGISTER_USER_CERTIFICATE, async (csr: string) => {
-      if (!this._permsData) {
-        console.log('NO PERMS DATA')
-        return
-      }
-      await this.registerUser(csr)
-    })
-    this.setRouting()
-  }
-
-  public setCertificates(certs: string[]) {
-    this.certificates = certs
-  }
-
-  private pendingPromise: Promise<RegistrarResponse> | null = null
-
-  private setRouting() {
-    // @ts-ignore
-    this._app.use(express.json())
-    this._app.post('/register', async (req, res): Promise<void> => {
-      if (this.pendingPromise) return
-      this.pendingPromise = this.registerUser(req.body.data)
-      const result = await this.pendingPromise
-      if (result) {
-        res.status(result.status).send(result.body)
+    this.on(
+      RegistrationEvents.REGISTER_USER_CERTIFICATE,
+      async (payload: { csrs: string[]; certificates: string[] }) => {
+        // Lack of permsData means that we are not the owner of the community
```

Comment:
Only after manipulating the source code - as well as many many other things in code, practically you could add this kind of note to every method in the codebase - should it be our standard practice?

Reply:
I think it is helpful to add notes throughout the codebase whenever there is ambiguity or complexity that isn't completely obvious from the code. I think the Linux kernel is a great example of code that uses comments effectively: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/mm/zswap.c?id=ca56489c2f1df516801eeff06129d2e9e282ab7b#n632 ... it can be much easier for someone reading the code later to understand what is happening and why.

If having permsData doesn't unconditionally mean that we are the owner of the community, then we shouldn't label it as such. If there is more to the story, I think we should say so.

Security only comes from server-side operations. So we can't trust a client to always send us perfect data. Even if it requires them to rewrite their client-side code in order to do so. It's like a cashier not counting cash at the grocery store. In the p2p model, each peer is their own server. So before a peer accepts anything, they need to verify for themselves that it's acceptable.

Reply:
Ok that's good idea, I've added a comment